thetis/options.py

```diff
@@ -670,6 +670,31 @@ class ModelOptions2d(CommonModelOptions):
     use_supg_tracer = Bool(
         False, help="Use SUPG stabilisation in tracer advection").tag(config=True)
 
+    def __init__(self, *args, **kwargs):
+        self.tracer_metadata = {}
+        super(ModelOptions2d, self).__init__(*args, **kwargs)
+
+    def add_tracer_2d(self, label, name, filename, shortname=None, unit='-', source=None):
+        """
+        Define a 2D tracer field
+
+        :arg label: field label used internally by Thetis, e.g. 'tracer_2d'
+        :arg name: human readable name for the tracer field, e.g. 'Tracer concentration'
+        :arg filename: file name for outputs, e.g. 'Tracer2d'
+        :kwarg shortname: short version of name, e.g. 'Tracer'
+        :kwarg unit: units for field, e.g. '-'
+        :kwarg source: associated source term
+        """
+        assert ' ' not in label, "Labels cannot contain spaces"
+        assert ' ' not in filename, "Filenames cannot contain spaces"
+        self.tracer_metadata[label] = {
+            'name': name,
+            'shortname': shortname or name,
+            'unit': unit,
+            'filename': filename,
+            'source': source,  # TODO: Validate input
+        }
```

Comment:
I guess another way would be to use nested options (like we do for timestepper_options for example). Then each entry would be a traitlet and we could validate the input (e.g. source). The API could look like:
```python
options.add_tracer(label)
options.tracer_options[label].source = foo
```
But the nested options can be a bit cumbersome to use in practice. I'd be happy to use this as it is for now. Maybe @stephankramer has some thoughts?

Reply:
That's a good point. I'll merge for now and leave it as a `TODO`. Thanks for the review!